lib/torrent.js

```diff
@@ -820,28 +821,49 @@ class Torrent extends EventEmitter {
     return newPeer
   }
 
-  addWebSeed (url) {
+  addWebSeed (urlOrConnection) {
     if (this.destroyed) throw new Error('torrent is destroyed')
 
-    if (!/^https?:\/\/.+/.test(url)) {
-      this.emit('warning', new Error(`ignoring invalid web seed: ${url}`))
-      this.emit('invalidPeer', url)
-      return
-    }
+    let id
+    let conn
+    if (typeof urlOrConnection === 'string') {
+      const url = urlOrConnection
+      id = urlOrConnection
+
+      if (!/^https?:\/\/.+/.test(url)) {
+        this.emit('warning', new Error(`ignoring invalid web seed: ${id}`))
+        this.emit('invalidPeer', id)
+        return
+      }
+
+      if (this._peers[id]) {
+        this.emit('warning', new Error(`ignoring duplicate web seed: ${id}`))
+        this.emit('invalidPeer', id)
+        return
+      }
+
+      conn = new WebConn(url, this)
+    } else if (urlOrConnection && typeof urlOrConnection.connId === 'string') {
```

Comment:
I don't think that checking for a `connId` property will correctly detect the default `WebConn` objects. Is there a better way to detect instances of `Wire`? Could check for a property on `Wire`, like `peerId` as a sanity check?